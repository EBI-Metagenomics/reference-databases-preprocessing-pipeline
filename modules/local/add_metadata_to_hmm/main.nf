process ADD_METADATA_TO_HMM {
    input:
    path ko_hmm_dir
    val ko_list

    output:
    path "*.hmm", emit: modified_ko_hmm

    script:
    """
    add_metadata_to_hmm.py ${ko_list} ${ko_hmm_dir}
    """
}