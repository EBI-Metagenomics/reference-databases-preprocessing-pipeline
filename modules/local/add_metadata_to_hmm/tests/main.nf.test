nextflow_process {

    name "Test Process ADD_METADATA_TO_HMM"
    script "../main.nf"
    process "ADD_METADATA_TO_HMM"

    test("Run module on the compressed test data") {

        when {
            process {
                """
                input[0] = file("${moduleDir}/tests/data/profiles.tar.gz", checkIfExists: true)
                input[1] = file("${moduleDir}/tests/data/ko_list.gz", checkIfExists: true)
                """
                
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Run module on the uncompressed test data") {

        when {
            process {
                """
                input[0] = file("${moduleDir}/tests/data/profiles", checkIfExists: true)
                input[1] = file("${moduleDir}/tests/data/ko_list", checkIfExists: true)
                """
                
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
