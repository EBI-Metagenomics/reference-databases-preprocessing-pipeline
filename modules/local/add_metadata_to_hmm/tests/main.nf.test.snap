{
    "Run ADD_METADATA_TO_HMM module on uncompressed test data": {
        "content": [
            {
                "0": [
                    "kofam_modified:md5,b209e0e2fb52685c67889e3dcaff14ff"
                ],
                "1": [
                    "versions.yml:md5,8f88ca94440f324703ce2f8126af124d"
                ],
                "kofam_modified": [
                    "kofam_modified:md5,b209e0e2fb52685c67889e3dcaff14ff"
                ],
                "versions": [
                    "versions.yml:md5,8f88ca94440f324703ce2f8126af124d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-25T15:41:19.195888"
    },
    "Stub mode": {
        "content": [
            {
                "0": [
                    "kofam_modified:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "versions.yml:md5,8f88ca94440f324703ce2f8126af124d"
                ],
                "kofam_modified": [
                    "kofam_modified:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,8f88ca94440f324703ce2f8126af124d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-25T15:41:22.015144"
    },
    "Run ADD_METADATA_TO_HMM module on compressed test data": {
        "content": [
            {
                "0": [
                    "kofam_modified:md5,b209e0e2fb52685c67889e3dcaff14ff"
                ],
                "1": [
                    "versions.yml:md5,8f88ca94440f324703ce2f8126af124d"
                ],
                "kofam_modified": [
                    "kofam_modified:md5,b209e0e2fb52685c67889e3dcaff14ff"
                ],
                "versions": [
                    "versions.yml:md5,8f88ca94440f324703ce2f8126af124d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-25T15:41:16.352569"
    }
}